File: backend/app/schemas/assets.py
import uuid
from datetime import datetime, date
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel


class AssetPhotoSchema(BaseModel):
    id: uuid.UUID
    s3_key: str
    original_filename: Optional[str] = None
    is_primary: bool

    model_config = {"from_attributes": True}


class AssetSchema(BaseModel):
    id: uuid.UUID
    inventory_number: str
    barcode: Optional[str] = None
    name: str
    description: Optional[str] = None
    category_id: Optional[int] = None
    department_id: Optional[int] = None
    room_id: Optional[int] = None
    initial_cost: Optional[Decimal] = None
    residual_value: Optional[Decimal] = None
    commissioning_date: Optional[date] = None
    useful_life_months: Optional[int] = None
    status: str
    inventory_status: str
    one_c_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    last_scanned_at: Optional[datetime] = None
    photos: list[AssetPhotoSchema] = []

    model_config = {"from_attributes": True}


class AssetCreateSchema(BaseModel):
    inventory_number: str
    barcode: Optional[str] = None
    name: str
    description: Optional[str] = None
    category_id: Optional[int] = None
    department_id: Optional[int] = None
    room_id: Optional[int] = None
    initial_cost: Optional[Decimal] = None
    residual_value: Optional[Decimal] = None
    commissioning_date: Optional[date] = None
    useful_life_months: Optional[int] = None
    one_c_id: Optional[str] = None


class AssetUpdateSchema(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    barcode: Optional[str] = None
    category_id: Optional[int] = None
    department_id: Optional[int] = None
    room_id: Optional[int] = None
    status: Optional[str] = None

File: admin/src/api/assets.ts
import { api } from './client'

export interface Asset {
  id: string
  inventory_number: string
  barcode: string | null
  name: string
  description: string | null
  category_id: number | null
  department_id: number | null
  room_id: number | null
  initial_cost: string | null
  residual_value: string | null
  commissioning_date: string | null
  useful_life_months: number | null
  status: string
  inventory_status: string
  one_c_id: string | null
  created_at: string
  updated_at: string
  last_scanned_at: string | null
  photos: { id: string; s3_key: string; is_primary: boolean }[]
}

export interface AssetCreate {
  inventory_number: string
  name: string
  barcode?: string
  description?: string
  category_id?: number
  department_id?: number
  room_id?: number
  initial_cost?: number
  one_c_id?: string
}

export interface AssetUpdate {
  name?: string
  description?: string
  barcode?: string
  status?: string
  category_id?: number
  department_id?: number
  room_id?: number
}

export const assetsApi = {
  list: (params?: { page?: number; limit?: number; search?: string; status?: string }) =>
    api.get<Asset[]>('/assets', { params }),

  get: (id: string) => api.get<Asset>(`/assets/${id}`),

  create: (data: AssetCreate) => api.post<Asset>('/assets', data),

  update: (id: string, data: AssetUpdate) => api.patch<Asset>(`/assets/${id}`, data),

  delete: (id: string) => api.delete(`/assets/${id}`),
}
